internal/migrations/changeset_listener.go

```diff
@@ -0,0 +1,272 @@
+package migrations
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/kwilteam/kwil-db/common"
+	"github.com/kwilteam/kwil-db/core/adminclient"
+	"github.com/kwilteam/kwil-db/core/log"
+	"github.com/kwilteam/kwil-db/extensions/listeners"
+)
+
+// Changeset Extension polls the changesets from the old chain  during migration.
+// The changesets received are broadcasted to the eventstore under ChangesetMigrationEvent type for voting.
+// The changesets are then applied to the new chain's database once the resolution is approved.
+const (
+	ListenerName = "migrations"
+	retryDelay   = 1 * time.Second // polling frequency for changesets
+)
+
+func init() {
+	// Register the listener with the name "migrations"
+	err := listeners.RegisterListener(ListenerName, Start)
+	if err != nil {
+		panic(err)
+	}
+}
+
+type MigrationConfig struct {
+	// StartHeight is the block height at which the migration started on the old chain.
+	StartHeight int64
+
+	// End height is the block height at which the migration ended on the old chain.
+	EndHeight int64
+
+	// AdminListenAddress is the address of the admin server to receive changesets from.
+	AdminListenAddress string
+
+	// AdminPass is the admin server password.
+	AdminPass string
+
+	// KwildTLSCertFile is the path to the TLS certificate file for the Kwil node.
+	// The path should be an absolute path or relative to the node's root directory.
+	KwildTLSCertFile string
```

Comment:
Can we further automate this? Essentially, just give the new node an endpoint to connect to, and it will contact that node, read the migration metadata, and apply it all itself?